### ISSUE TITLE
Formulas not displayed correctly in Sphinx

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#8529]
#### Description of the problem

Formulas don't render correctly any more.  IIRC they worked fine a week or two ago.  For example, in my locally built version of the doc in ..../coq/doc/sphinx/_build/html/language/cic.html#typing-rules:

![image](https://user-images.githubusercontent.com/1253341/45920522-fe24ee00-be59-11e8-94e4-2d49137fa0c0.png)

@cpitclaudel do you know anything about this?

**Comment by Zimmi48** [comment]
More precisely, this is a regression that comes with the PDF manual PR:
- here is this chapter right before this PR was backported to v8.8: https://coq.gitlab.io/-/coq/-/jobs/99619717/artifacts/_install_ci/share/doc/coq/sphinx/html/language/cic.html
- here it is right after: https://coq.gitlab.io/-/coq/-/jobs/100292516/artifacts/_install_ci/share/doc/coq/sphinx/html/language/cic.html

This is critical enough to be a blocker for 8.8.2 :cry:

**Comment by Zimmi48** [comment]
Another minor regression is the location of the first footnote on this page: https://coq.gitlab.io/-/coq/-/jobs/100292516/artifacts/_install_ci/share/doc/coq/sphinx/html/index.html
I forgot to check the rendered output before merging even if I had said I wanted to.

**Comment by Zimmi48** [comment]
Another minor thing I'd like to fix at the same time is to use `refman-pdf` and `refman-html` as the name of the targets and preserve the previous meaning of `sphinx` as a synonymous of `refman-html`.

**Comment by Zimmi48** [comment]
@cpitclaudel The first problem mentioned seems to be that the new implementation of `.. preamble::` simply does not work.

**Comment by cpitclaudel** [comment]
I'll look at it right away.  Thanks for the ping.

**Comment by cpitclaudel** [comment]
I'm working on this, but I'm having trouble building:

```
------------
Running 'internal xindy -L english -C utf8  -M sphinx.xdy -I xelatex  -o "CoqRefMan.ind" "CoqRefMan.idx"'
------------
Latexmk: calling xindy( -L english -C utf8 -M sphinx.xdy -I xelatex -o CoqRefMan.ind CoqRefMan.idx )
Unsupported input markup xelatex.

usage: xindy [-V?h] [-qv] [-d magic] [-o outfile.ind] [-t log] \
            [-L lang] [-C codepage] [-M module] [-I input] \
            [--interactive] [--mem-file xindy.mem] \
	    [idx0 idx1 ...]

GNU-STYLE LONG OPTIONS FOR SHORT OPTIONS:

 -V / --version
 -? / -h / --help
 -q / --quiet
 -v / --verbose
 -d / --debug          (multiple times)
                       (supported: script, keep_tmpfiles, markup, level=n)
 -o / --out-file
 -t / --log-file
 -L / --language
 -C / --codepage
 -M / --module         (multiple times)
 -I / --input-markup   (supported: latex, omega, xindy)
```

I installed `xindy` from the Ubuntu repositories:

```
$ xindy -V
xindy release: unknown
xindy script version: 1.16
xindy kernel version: 3.0
CLISP version 2.49 (2010-07-07) (built on brownie.buildd [127.0.1.1])
    architecture: X86_64
```

**Comment by Zimmi48** [comment]
You can't build the PDF version of the reference manual anymore? That's really weird. We didn't change much since you submitted your PR and I suppose it was working at the time, wasn't it?
Isn't your problem that you are missing the `xelatex` dependency? Are you using the same machine as you were using when you last worked on the PDF manual? I personally have texlive-full (installed with Nix) on my machine and no trouble building the doc.

**Comment by cpitclaudel** [comment]
The problem seems to be with Sphinx 1.8.0 :/ I'll investigate further after fixing this issue with formulae.

**Comment by Zimmi48** [comment]
There's one last regression with the PDF manual, which is that `install-doc-sphinx` disappeared. I recommend keeping it as it was before and adding `install-doc-refman`, `install-doc-refman-html` and `install-doc-refman-pdf`. I'm sorry for not having noticed all these issues before. I think that as I was finally involved in the PR to make it move quicker and I became frustrated with all the hurdles, I was less careful when reviewing it then.

**Comment by cpitclaudel** [comment]
> There's one last regression with the PDF manual, which is that install-doc-sphinx disappeared. I recommend keeping it as it was before and adding install-doc-refman, install-doc-refman-html and install-doc-refman-pdf. 

I think we briefly discussed this.  I'm happy to fix this, but I'm puzzled by that code: none of the `install-doc-*` targets have dependencies, so I'm not sure how to write the new rules.

That peculiarity dates back to this:

```
commit 586d6add6d2d4405eacf9241ee0d55f58435e364
Author: notin <notin@85f007b7-540e-0410-9357-904b9bb8a0f7>
Date:   Fri Oct 17 14:06:04 2008 +0000

    Suppression de la dépendance de install-doc envers doc :
      - make install ne compile plus la doc
      - make world compile la doc (sauf si option --with-doc no)

    git-svn-id: svn+ssh://scm.gforge.inria.fr/svn/coq/trunk@11460 85f007b7-540e-0410-9357-904b9bb8a0f7
```

**Comment by Zimmi48** [comment]
So this is probably linked to this:

https://github.com/coq/coq/blob/fb8365e96ccb5d25d3810af99c36e7b27fec8fad/Makefile.doc#L15-L19

`make documentation` is run in every case, and probably `make install-doc` as well. This seems crazy but for now we should probably preserve this behavior. There's the plan to get rid of `configure` entirely when we move to a full Dune-based build system.

**Comment by Zimmi48** [comment]
@cpitclaudel If you manage to fix the problem with Sphinx 1.8.0 today this would be great. Otherwise, too bad for 8.8.2 but the release of this version cannot be delayed even more.

**Comment by cpitclaudel** [comment]
> @cpitclaudel If you manage to fix the problem with Sphinx 1.8.0 today this would be great. Otherwise, too bad for 8.8.2 but the release of this version cannot be delayed even more.

Right, I don't think we should hold anything up for the build issue that I ran into :)
